UMN-VMWareRA.psm1

```diff
@@ -1197,6 +1197,321 @@ function Set-VMWareRAVMTagAttachedToObject {
+
+    .PARAMETER clusterID
+        ID of cluster to return list of VMs from (able to return more then 1000 VMs.) Overrides cluster parameter if specified.
+
+    .OUTPUTS
+        List of VM objects based on parameters specified.
+
+    .Notes
+        Author: Aaron Smith
+#>
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory)]
+        [string]$vCenter,
+        
+        [Parameter(Mandatory)]
+        [string]$sessionID,
+
+        [string]$name,
+
+        [string]$host,
+
+        [string]$hostID,
+
+        [string]$cluster,
+
+        [string]$clusterID
+    )
+
+    Begin
+    {
+    }
+    Process
+    {
+        if ($cluster)
+        {
+            $clusterID = (Get-VMWareRACluster -vCenter $vCenter -sessionID $sessionID -name $cluster).cluster;
+
+            if (!$clusterID)
+            {
+                throw "Cluster ${cluster} not found by name";
+            }
+        }
+
+        if ($host)
+        {
+            $hostID = (Get-VMWareRAHost -vCenter $vCenter -sessionID $sessionID -name $host).host;
+
+            if (!$hostID)
+            {
+                throw "VMHost ${host} not found by name";
+            }
+        }
+
+        ## List of VMs compiled from queries for return
+        [Array] $vmList = @();
+
+        ## Construct url
+        $url = "https://$vCenter/rest/vcenter/vm"
+
+        [Array] $urlFilters = @();
+
+        if ($hostID)
+        {
+            $urlFilters += "filter.hosts=${hostID}";
+        }
+
+        if ($urlFilters.Count -gt 0)
+        {
+            $url += "?";
+
+            foreach ($urlFilterItem in $urlFilters)
+            {
+                $url += $urlFilterItem + "&";
+            }
+
+            $return = Invoke-WebRequest -Uri $url -Method Get -Headers @{'vmware-api-session-id'=$sessionID} -ContentType 'application/json' -UseBasicParsing
+            $vmList += (($return.Content | ConvertFrom-Json).value)
+        }
+        else
+        {
+            [Array] $hostList = @();
+
+            if ($clusterID)
+            {
+                $hostList += Get-VMWareRAHost -vCenter $vCenter -sessionID $sessionID -clusterID $clusterID;
+            }
+            else
+            {
+                $hostList += Get-VMWareRAHost -vCenter $vCenter -sessionID $sessionID;
+            }
+
+            foreach ($hostItem in $hostList)
+            {
+                $hostID = $hostItem.host;
+                $url = "https://$vCenter/rest/vcenter/vm?filter.hosts=${hostID}"
+
+                $return = Invoke-WebRequest -Uri $url -Method Get -Headers @{'vmware-api-session-id'=$sessionID} -ContentType 'application/json' -UseBasicParsing
+                $vmList += (($return.Content | ConvertFrom-Json).value);
+            }
+        }
+
+        if ($name)
+        {
+            [Array] $filteredVMList = @();
+
+            foreach ( $vmItem in $vmList )
+            {
+                if ($vmItem.name -like "*${name}*")
```

Comment:
The issue with this is that you could get back multiple results when maybe all you wanted was to search without case sensitivity as you have forced in the asterisk.  Remove those and it will match an exact name but not case-sensitive.  Let the use call the function with asterisks in the $name if that is the desired result.  This should mirror power-cli.

Reply:
Makes sense.  I'll make this change.